### DISCUSSION TITLE
Filtering in the filesystem routes

### DISCUSSION THREAD

**Discussion by NobbZ** [gatsbyjs/gatsby#38270] [Help] [answered]
I have 2 different filesystem sources and MDX plugin, roughly like this:

```ts
    {
      resolve: "gatsby-source-filesystem",
      options: {
        name: "blog",
        path: `${__dirname}/blog`,
      },
    },
    {
      resolve: "gatsby-source-filesystem",
      options: {
        name: "author",
        path: `${__dirname}/author`,
      },
    },
    "gatsby-plugin-mdx",
    "gatsby-plugin-mdx-source-name",
```

I could probably use the `createPages` node-API, though I would prefer to be able to use file system routing as much as possible, as it is easier to find a pages source that way.

But there doesn't seem to be a way to filter the nodes for the implied query.

Is there a way to have the routes `{mdx.frontmatter__date}-{mdx.frontmatter__slug}/index.tsx` and `author/{mdx.frontmatter__slug}/index.ts` to coexist, without the latter being called with blog posts?

I hoped I could use the `source` as provided by `gatsby-plugin-mdx-source-name` which I can use to filter in the corresponding index pages just fine, though I haven't yet found a way to filter on that fields value in the routes.

And as I do not want to have the blog posts nested in a folder, I can't route on `{mdx.fields__source}` either, especially not as the blogs and authors use a different expansion schema.

**Comment by NobbZ** [comment]
After I put some thoughts into it, it feels as if a local transformer, that transforms Mdx-nodes into "author" and "blog" nodes, and perhaps others later on, might be the cleanest solution.

Is this indeed a viable solution?

I hope I will be able to implement a MVP in the next days, unless someone here has a better idea.

**Reply by LekoArts** [reply, marked as answer]
Yes, you're on the right path. The File System Route API doesn't allow filtering and as such if you want to split the content, you'll need to create individual interfaces for them.

Here's where the `source` check is happening: https://github.com/LekoArts/gatsby-themes/blob/main/themes/gatsby-theme-jodie-core/gatsby-node.mjs#L127-L130

You can take the whole https://github.com/LekoArts/gatsby-themes/blob/main/themes/gatsby-theme-jodie-core/gatsby-node.mjs file as an example on how to achieve what you want
